### PR TITLE
Redo linalg_ext.scatter to support explicitly defined index update sizes

### PR DESCRIPTION
**Pull request by rsuderman**
Scatter can be performed by now allowing additional dimensions on the update
tensor. This supports the existing tiling method but requires slight
modifications to validators and the scalar computation.

### CODE REVIEW

**Comment by GMNGeoffrey** [comment]
King Linalg III

linalgIII

**Review by hanhanW — COMMENTED** [review]
Nice work! I'll review the change soon.

Can you update op description as well?

https://github.com/google/iree/blob/db64daa4006abb0ccd3474a9791f7df140aa0074/llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td#L50-L75

File: llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp [hunk 1]
```diff
@@ -136,20 +136,49 @@ static LogicalResult verifyScatterOp(ScatterOp op) {
...
-        originalType.getDimSize(dim)) {
+
+  // indexDepth + update dims should cover the original dims. The first dim of
+  // update is the number of updates.
+  if (originalType.getRank() > indexDepth + updateType.getRank() - 1) {
+    return op.emitOpError(
+        "index depth and update value does not cover rank of original value");
+  }
+
+  // Validate the non-indexed update dims are
```
Comment (hanhanW) [inline llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp:151]: The sentence is not finished?

**Comment by MaheshRavishankar** [comment]
> King Linalg III
> 
> linalgIII

More like "King Scatter the X"

**Comment by rsuderman** [comment]
> > King Linalg III
> > linalgIII
> 
> More like "King Scatter the X"

How about Vlad III, also known as Vlad the Scatterer?

**Review by hanhanW — APPROVED** [review]

**Pull request merged**